### PR TITLE
feat(thread)!: support `newly_created`, add `on_thread_create` event

### LINKED ISSUE #323: Track thread.newly_created
### Summary

implement api info to more easily tell thread events apart

### What is the feature request for?

The core library

### The Problem

Its not possible to tell exactly what the thread event is at the moment, whether the bot was added to the thread or the thread was just created.

### The Ideal Solution

Implement https://github.com/discord/discord-api-docs/commit/1875d80011ea62bf555ce807459ccf97950291e9

### The Current Solution

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by shiftinv**
## Summary

Changes the `THREAD_CREATE` handling to dispatch a `thread_create` instead of `thread_join` event according to the `newly_created` field in the payload.

closes #323

ref: https://github.com/discord/discord-api-docs/commit/1875d80011ea62bf555ce807459ccf97950291e9

Previous attempt was #331, which partially changed how thread caching works. While this is certainly something that should be fixed, it should really be done separately.

## Checklist

- [x] If code changes were made, then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [x] This PR adds something new (e.g. new method or parameters)
- [x] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: docs/api.rst [hunk 1]
```diff
@@ -812,18 +812,41 @@ to handle it, which defaults to print a traceback and ignoring the exception.
...
 
+    .. versionchanged:: 2.5
+        This is no longer being called when a thread is created, unless the bot itself created
+        the thread (since the thread creator automatically becomes a thread member).
+        See :func:`on_thread_create` instead.
+
     :param thread: The thread that got joined.
     :type thread: :class:`Thread`
 
+.. function:: on_thread_create(thread)
```
Comment (onerandomusername) [inline docs/api.rst:832]: should be above the docs for on_thread_delete imo

**Review by onerandomusername — COMMENTED** [review]
one comment, other than that, looks good

File: docs/api.rst (hunk 1 above)
Comment (shiftinv) [reply docs/api.rst]: ce811dce5731bd89e143bd0dfb45b6e232ccf6c9

**Review by onerandomusername — APPROVED** [review]
lgtm

**Pull request merged**